### PR TITLE
UI for onboarding flow

### PR DESCRIPTION
**Pull request by Elanhant**
This PR adds pages and components for most of the onboarding flow. No API calls or anything, with the exception of calling `useSites` to display a list of detected sites.

To get to the onboarding flow, you can use a temporary "Onboarding" button on the homepage:
Screen Shot 2020-08-27 at 12 48 09 PM

### CODE REVIEW

**Review by ascorbic — COMMENTED** [review]
image
image

I don't know if this is due to an existing issue, but the layout doesn't handle smaller window sizes very well. This is at the default window size.

**Comment by Elanhant** [comment]
> I don't know if this is due to an existing issue, but the layout doesn't handle smaller window sizes very well. This is at the default window size.

Interesting, I think it was working fine for me, but I'll make sure to check again!

**Comment by Elanhant** [comment]
Onboarding flow should work better now on smaller screens (turns out having `position: absolute` on that SVG graphic was the culprit, I changed it to `position: fixed`).

File: src/components/folder-name.tsx [hunk 1]
```diff
@@ -24,7 +24,9 @@ export function FolderName({ sitePath }: IProps): JSX.Element {
         alignItems: `center`,
       }}
     >
-      <MdFolderOpen size={12} /> <span sx={{ paddingLeft: 2 }}>{folder}</span>
+      <MdFolderOpen size={12} aria-label="Folder:" />
+      {` `}
```
Comment (ascorbic) [inline src/components/folder-name.tsx:28]: I think this is a Prettier artefact

File: src/components/site-card/logs-viewer.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      }),
+    []
+  )
+
+  return (
+    <StyledPanel>
+      <StyledPanelHeader onCloseButtonClick={onDismiss}>Logs</StyledPanelHeader>
+      <RawLogs
+        // RawLogs expect "message" to be a string but in factthey also support elements
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
Comment (ascorbic) [inline src/components/site-card/logs-viewer.tsx:26]: So this is a workaround for bad typings in gatsby-interface? Can this be fixed there?

**Comment by ascorbic** [comment]
This fixes the horizontal scroll, but there's still a vertical scroll at the default window size. I think this is because the padding causes the grid to switch to single column at the default window size. Could this be fixed?
image

File: src/pages/onboarding/[step].tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      <main
+        sx={{
+          overflowY: `auto`,
+          px: `15%`,
+          pb: 10,
+          pt: [12, null, null, null, `10rem`],
+          fontFamily: `sans`,
+        }}
+      >
+        {step === `editor` && (
```
Comment (ascorbic) [inline src/pages/onboarding/[step].tsx]: I'm not convinced of the value of using unified routes in this situation. There are a fixed set of steps, each with its own component. I'd rather create a wrapper component to share, and a page template for each step

**Comment by Elanhant** [comment]
> This fixes the horizontal scroll, but there's still a vertical scroll at the default window size. I think this is because the padding causes the grid to switch to single column at the default window size. Could this be fixed?

We should probably consult with design on this. I also have a feeling that our current default size is too small, perhaps we can just increase it if possible?

**Comment by ascorbic** [comment]
> We should probably consult with design on this. I also have a feeling that our current default size is too small, perhaps we can just increase it if possible?

I agree, but I do think the responsive behaviour could be improved, so that the top and left padding decreases. I'm happy to levae it for now, and will increase the default size so that it fits iniitally.

**Pull request merged**